src/planner/binder/statement/bind_copy.cpp

```diff
@@ -185,6 +192,7 @@ BoundStatement Binder::BindCopyTo(CopyStatement &stmt) {
 	copy->rotate = rotate;
 	copy->partition_output = !partition_cols.empty();
 	copy->partition_columns = std::move(partition_cols);
+	copy->return_files = return_files;
```

Comment:
The copy statement should return `StatementReturnType::QUERY_RESULT` when return files is set - this way we signal the (correct) return type which is a result and not just the number of changed rows.